### PR TITLE
Accessibility for Font size/family and Heading dropdowns in toolbar

### PR DESCRIPTION
**Pull request by mmotyczynska**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (font, heading): Screen readers should now announce selected option in dropdown lists for font size, font family and heading. Closes https://github.com/ckeditor/ckeditor5/issues/13250.

---

### Additional information

There was already [one solution](https://github.com/ckeditor/ckeditor5/pull/13749) in which we tried to implement this but it was not good enough. As described [here](https://github.com/ckeditor/ckeditor5/pull/13749#issuecomment-1507057330), we decided to go with another approach.
Besides font we added here also heading feature to make sure that this time the proposed solution will cover also dropdowns with the selected value shown in the toolbar.

### CODE REVIEW

**Review by illia-stv — APPROVED** [review]
For me it looks good!!!

File: packages/ckeditor5-ui/src/button/buttonview.ts [hunk 1]
```diff
@@ -140,6 +140,26 @@ export default class ButtonView extends View<HTMLButtonElement> implements Butto
...
 
+	/**
+	 * @inheritDoc
+	 */
+	declare public role: string | undefined;
+
+	/**
+	 * @inheritDoc
+	 */
+	declare public ariaChecked: boolean | undefined;
```
Comment (oleq) [inline packages/ckeditor5-ui/src/button/buttonview.ts:151]: Don't we need these in the `Button` interface as well?

File: packages/ckeditor5-ui/src/button/buttonview.ts [hunk 2]
```diff
@@ -209,10 +233,13 @@ export default class ButtonView extends View<HTMLButtonElement> implements Butto
 					bind.if( 'withText', 'ck-button_with-text' ),
 					bind.if( 'withKeystroke', 'ck-button_with-keystroke' )
 				],
+				role: bind.to( 'role' ),
 				type: bind.to( 'type', value => value ? value : 'button' ),
 				tabindex: bind.to( 'tabindex' ),
-				'aria-labelledby': `ck-editor__aria-label_${ ariaLabelUid }`,
+				'aria-label': bind.to( 'ariaLabel' ),
+				'aria-labelledby': bind.to( 'ariaLabelledBy' ),
```
Comment (oleq) [inline packages/ckeditor5-ui/src/button/buttonview.ts:240]: The existing test checks if `aria-labelledby` starts with `ck-editor__aria-label`. This is a default value but we also need a test to see if the new observable `ariaLabelledBy` property is actually observable and what it does.

File: packages/ckeditor5-ui/src/dropdown/utils.ts [hunk 3]
```diff
@@ -310,6 +310,7 @@ export function addListToDropdown(
 	itemsOrCallback: Collection<ListDropdownItemDefinition> | ( () => Collection<ListDropdownItemDefinition> ),
 	options: {
 		ariaLabel?: string;
+		role?: string;
```
Comment (oleq) [inline packages/ckeditor5-ui/src/dropdown/utils.ts:313]: Missing API docs.

File: packages/ckeditor5-heading/src/headingui.ts [hunk 4]
```diff
@@ -83,12 +84,17 @@ export default class HeadingUI extends Plugin {
...
 			const dropdownView = createDropdown( locale );
-			addListToDropdown( dropdownView, itemDefinitions );
+			addListToDropdown( dropdownView, itemDefinitions, {
+				ariaLabel: accessibleLabel,
+				role: 'menu'
+			} );
 
 			dropdownView.buttonView.set( {
+				ariaLabel: accessibleLabel,
+				ariaLabelledBy: undefined,
```
Comment (oleq) [inline packages/ckeditor5-heading/src/headingui.ts:94]: These changes have no test coverage.

File: packages/ckeditor5-ui/src/button/buttonview.ts (hunk 2 above)
Comment (mmotyczynska) [reply packages/ckeditor5-ui/src/button/buttonview.ts:240]: Test added.

**Pull request merged**